src/main/kotlin/org/rust/lang/utils/RsDiagnostic.kt

```diff
@@ -51,22 +52,34 @@ sealed class RsDiagnostic(
                 fixes = buildList {
                     if (expectedTy is TyNumeric && isActualTyNumeric()) {
                         add(AddAsTyFix(element, expectedTy))
-                    } else if (isFromActualImplForExpected(element)) {
-                        add(ConvertToTyUsingFromTraitFix(element, expectedTy))
+                    } else  if (element is RsElement){
+                        val items = StdKnownItems.relativeTo(element)
+                        val lookup = ImplLookup(element.project, items)
+                        if (isFromActualImplForExpected(items, lookup)) {
+                            add(ConvertToTyUsingFromTraitFix(element, expectedTy))
+                        }
+                        if (isToOwnedImplWithExcpectedForActual(items, lookup)) {
+                            add(ConvertToOwnedTyFix(element))
+                        }
                     }
                 }
             )
         }
 
         private fun isActualTyNumeric() = actualTy is TyNumeric || actualTy is TyInfer.IntVar || actualTy is TyInfer.FloatVar
 
-        private fun isFromActualImplForExpected(element: PsiElement): Boolean {
-            val items = StdKnownItems.relativeTo(element as? RsElement ?: return false)
-            val lookup = ImplLookup(element.project, items)
+        private fun isFromActualImplForExpected(items: StdKnownItems, lookup: ImplLookup): Boolean {
             val fromTrait = items.findFromTrait() ?: return false
             return lookup.select(TraitRef(expectedTy, fromTrait.withSubst(actualTy))).ok() != null
         }
 
+        private fun isToOwnedImplWithExcpectedForActual(items: StdKnownItems, lookup: ImplLookup): Boolean {
+            val toOwnedTrait = items.findToOwnedTrait() ?: return false
+            val declaredTy = lookup.select(TraitRef(actualTy, BoundElement(toOwnedTrait))).ok()?.impl?.members
```

Comment:
Let's use `selectProjection` here. It is more generic API for associated type selection. I can't quickly imagine a case where `selectProjection` works better right now, but it is more clear and in future it should work, for example, with associated type bindings:
```rust
fn foo<T: ToOwned<Owned=X>>(t: T) { bar(t); }
```

Reply:
maybe done, not sure i used the selectProjection the best/right way

Reply:
Hmm, it can look a bit cleaner if we extract `toOwnedTrait.associatedTypesTransitively.find { it.name == "Owned" }` to a separate function (or make `findAssociatedType` in `ImplLookup` public and just use it), but it is something minor.